### PR TITLE
patched for Zoneminder 1.24+

### PR DESCRIPTION
**Pull request by Brawn1**
changes
* changed script to scan folder instead of links.
* added the possibility to work on host with docker containers
* added feature to check if path to delete is in zmcache path

### CODE REVIEW

File: zoneminder/zoneminder-event-cleanup.sh [hunk 1]
```diff
@@ -4,23 +4,31 @@
 # Purpose:	Clean up old Zoneminder events but exclude archived events
 # Authors:	Claudio Kuenzler (2018)
 # Doc:		https://www.claudiokuenzler.com/blog/814/how-to-manually-clean-up-delete-zoneminder-events
-# History:	
-# 2018-12-14 First version
+# History:
+# 2020-08-06 changed for zoneminder 1.24+ (@Brawn1)
+# 2020-07-24 added Docker Mysql (@Brawn1)
```
Comment (Napsty) [inline zoneminder/zoneminder-event-cleanup.sh]: Please append newer change log below, in style 
older
newer

File: zoneminder/zoneminder-event-cleanup.sh [hunk 2]
```diff
@@ -4,23 +4,31 @@
...
 # 2018-12-18 DB Name variable in first query
+# 2018-12-14 First version
 ###########################################################################
 # User variables
-olderthan=2 # Defines the minimum age in days of the events to be deleted
-zmcache=/var/cache/zoneminder/events # Defines the path where zm stores events
+olderthan=7 # Defines the minimum age in days of the events to be deleted
```
Comment (Napsty) [inline zoneminder/zoneminder-event-cleanup.sh]: We should add a parameter to the script. I'm fine with a default of 7 days, but user should have the option to easily use another limit.

File: zoneminder/zoneminder-event-cleanup.sh [hunk 3]
```diff
@@ -4,23 +4,31 @@
...
+zmcache=/var/lib/docker/volumes/zoneminder_zmcache/_data/events # Defines the path where zm stores events
```
Comment (Napsty) [inline zoneminder/zoneminder-event-cleanup.sh]: The hard-coded default path should stay `/var/cache/zoneminder/events`, I honestly don't think there are more production Docker installations of ZM than real machines out there. Another parameter for easily defining another zm events path would be good here.

File: zoneminder/zoneminder-event-cleanup.sh [hunk 4]
```diff
@@ -4,23 +4,31 @@
...
+docker_mysql="" # if mysql running as docker container, add the mysql container name (example: docker_mysql=zoneminder_mysql_1), or leafe it blank
 
 # Fixed variables
 tmpfile=/tmp/$RANDOM
 
-# Get archived events from database
-declare -a archived=( $(mysql -N -u $mysqluser --password=$mysqlpass -e "select Id from ${mysqldb}.Events where Archived = 1;") )
+if [[ ! -z ${docker_mysql} ]]; then
+  # Get archived events from database
+  declare -a archived=( $(docker exec -it ${docker_mysql} bash -lc 'mysql -N -h ${mysqlhost} -u ${mysqluser} --password=${mysqlpass} -e "select Id from ${mysqldb}.Events where Archived = 1;"') )
```
Comment (Napsty) [inline zoneminder/zoneminder-event-cleanup.sh]: Don't see the point here. It doesn't matter if mysql runs on localhost or on another remote host. Communication happens over tcp. You can use the same `declare -a archived` line from your `else` statement to retrieve this. The magic in Docker-based ZM installations is probably to read the environment variable of the mysql host.

File: zoneminder/zoneminder-event-cleanup.sh [hunk 5]
```diff
@@ -30,20 +38,28 @@ for id in $(echo ${archived[*]}); do
 done
 
 # Find events according to our filter
-find ${zmcache}/ -mindepth 2 -type l -mtime +${olderthan} ${findexception[*]} -exec ls -la {} + > $tmpfile
+find ${zmcache}/ -mindepth 3 -maxdepth 3 -type d -mtime +${olderthan} ${findexception[*]} -exec ls -ld {} + > $tmpfile
```
Comment (Napsty) [inline zoneminder/zoneminder-event-cleanup.sh]: Looks good! Just double-checked with my Zoneminder 1.34 version and this should work. However another parameter could help users of "old installations" to continue use the script. Example: user uses the parameter `-o` (for old/outdated) and the script will use the old `find` command with the symlinks.

Comment (Brawn1) [reply zoneminder/zoneminder-event-cleanup.sh]: Hmm..., that's a good point.
i will update the script to use optional a variable to declare, it's for Zoneminder with Version 1.20 and lower.

File: zoneminder/zoneminder-event-cleanup.sh (hunk 3 above)
Comment (Brawn1) [reply zoneminder/zoneminder-event-cleanup.sh]: ja sorry, i see that after the Pull Request.
i change that today with the optional feature. :-)

File: zoneminder/zoneminder-event-cleanup.sh (hunk 4 above)
Comment (Brawn1) [reply zoneminder/zoneminder-event-cleanup.sh]: The intention was, to isolate the MySQL Instance on the host (Our System runs on docker with Video Cards and Net Cams).
but i will go back with the part to the old one and create a own Container, to use the script with docker.
So the script shrinks in size and is easier to use. ;-)

File: zoneminder/zoneminder-event-cleanup.sh (hunk 5 above)
Comment (Brawn1) [reply zoneminder/zoneminder-event-cleanup.sh]: i have added in the user settings *zm_linked_version*.
So we can now change the script to cleanup the old version von Zoneminder or the new one (default is blank).

**Comment by Napsty** [comment]
Successfully tested on ZM 1.34. Thank you!

**Pull request merged**